File: tools/packages/db_connection_redis/packages_py/db_connection_redis/src/db_connection_redis/schemas.py
from typing import Optional, Literal
from pydantic import BaseModel, Field

class RedisConfigValidator(BaseModel):
    """Validator for Redis configuration parameters."""
    host: str = Field(min_length=1)
    port: int = Field(ge=1, le=65535)
    username: Optional[str] = None
    password: Optional[str] = None
    db: int = Field(default=0, ge=0)
    unix_socket_path: Optional[str] = None
    use_ssl: bool = False
    ssl_cert_reqs: Literal["none", "optional", "required"] = "none"
    ssl_ca_certs: Optional[str] = None
    ssl_check_hostname: bool = False
    socket_timeout: float = Field(default=5.0, gt=0)
    socket_connect_timeout: float = Field(default=5.0, gt=0)
    retry_on_timeout: bool = False
    max_connections: Optional[int] = Field(default=None, gt=0)
    health_check_interval: float = Field(default=0, ge=0)

File: packages_mjs/db_connection_redis/src/schemas.mjs
import { z } from 'zod';
import { VALID_VENDORS } from './constants.mjs';

export const RedisConfigSchema = z.object({
    host: z.string().min(1),
    port: z.number().int().min(1).max(65535),
    username: z.string().nullable().optional(),
    password: z.string().nullable().optional(),
    db: z.number().int().min(0),
    useTls: z.boolean(),
    sslCertReqs: z.enum(['none', 'optional', 'required']),
    sslCaCerts: z.string().nullable().optional(),
    sslCheckHostname: z.boolean(),
    socketTimeout: z.number().positive(),
    socketConnectTimeout: z.number().positive(),
    retryOnTimeout: z.boolean(),
    maxConnections: z.number().int().positive().nullable().optional(),
    minConnections: z.number().int().min(0).nullable().optional(),
    healthCheckInterval: z.number().min(0),
    vendorType: z.enum(VALID_VENDORS).optional(),
});
